### ISSUE TITLE
Detailed requirements for training

### ISSUE THREAD

**Issue by dawson-miller** [RF5/simple-sashimi#1]
Can you please provide a detailed list of your training environment including versions of the packages?

While inference works, I am unable to train due to the following error.

CUDA extension for cauchy multiplication not found. Install by going to extensions/cauchy/ and running `python setup.py install`. This should speed up end-to-end training by 10-50%
Falling back on slow Cauchy kernel. Install at least one of pykeops or the CUDA extension for efficiency.
Initializing Training Process..
Traceback (most recent call last):
  File "/mnt/s/audio/simple-sashimi/train.py", line 350, in <module>
    main()
  File "/mnt/s/audio/simple-sashimi/train.py", line 319, in main
    str(OmegaConf.to_yaml(OmegaConf.structured(TrainConfig))))
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 125, in structured
    return OmegaConf.create(obj, parent, flags)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 178, in create
    return OmegaConf._create_impl(
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 900, in _create_impl
    format_and_raise(node=None, key=None, value=None, msg=str(e), cause=e)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 861, in _create_impl
    return DictConfig(
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 111, in __init__
    format_and_raise(node=None, key=key, value=None, cause=ex, msg=str(ex))
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 819, in format_and_raise
    _raise(ex, cause)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 94, in __init__
    self._set_value(content, flags=flags)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 647, in _set_value
    raise e
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 644, in _set_value
    self._set_value_impl(value, flags)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/dictconfig.py", line 674, in _set_value_impl
    data = get_structured_config_data(value, allow_objects=ao)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 495, in get_structured_config_data
    return get_dataclass_data(obj, allow_objects=allow_objects)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 421, in get_dataclass_data
    format_and_raise(
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 899, in format_and_raise
    _raise(ex, cause)
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/_utils.py", line 413, in get_dataclass_data
    d[name] = _maybe_wrap(
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1105, in _maybe_wrap
    return _node_wrap(
  File "/home/sys-island/venvs/torch_audio/lib/python3.10/site-packages/omegaconf/omegaconf.py", line 1087, in _node_wrap
    raise ValidationError(f"Unexpected type annotation: {type_str(ref_type)}")
omegaconf.errors.ValidationError: Unexpected type annotation: DistributedConfig
    full_key: distributed
    object_type=TrainConfig

**Comment by dawson-miller** [comment]
To add to my post, I have built environments against python version 3.8, 3.9, 3.10, and 3.11 and each one gives the same error message.

**Comment by RF5** [comment]
Sure, let me dump by env here. 

Python: 3.10.4
Packages:

```yaml
absl-py==1.0.0
aiohttp==3.8.1
aiosignal==1.2.0
antlr4-python3-runtime==4.8
anyio==3.5.0
appdirs==1.4.4
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
asttokens==2.0.5
astunparse==1.6.3
async-timeout==4.0.2
attrs==21.4.0
audioread==2.1.9
Babel==2.10.1
backcall==0.2.0
beautifulsoup4==4.11.1
bleach==5.0.0
cachetools==5.0.0
certifi==2020.6.20
cffi==1.15.0
chardet==5.0.0
charset-normalizer==2.0.12
click==8.1.2
commonmark==0.9.1
contourpy==1.0.6
cycler==0.11.0
datasets==2.1.0
debugpy==1.6.0
decorator==5.1.1
defusedxml==0.7.1
dill==0.3.4
einops==0.4.1
entrypoints==0.4
executing==0.8.3
fastjsonschema==2.15.3
fastprogress==1.0.2
filelock==3.6.0
flatbuffers==22.11.23
fonttools==4.32.0
frozenlist==1.3.0
fsspec==2022.3.0
gast==0.4.0
google-auth==2.6.5
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio==1.44.0
h5py==3.7.0
huggingface-hub==0.5.1
hydra-core==1.1.2
idna==3.3
inflect==5.6.0
ipykernel==6.13.0
ipython==8.2.0
ipython-genutils==0.2.0
ipywidgets==7.7.0
jedi==0.18.1
Jinja2==3.1.1
joblib==1.1.0
json5==0.9.6
jsonpickle==2.2.0
jsonschema==4.4.0
jupyter-client==7.2.2
jupyter-core==4.10.0
jupyter-server==1.16.0
jupyterlab==3.3.4
jupyterlab-pygments==0.2.2
jupyterlab-server==2.13.0
jupyterlab-widgets==1.1.0
keras==2.11.0
kiwisolver==1.4.2
libclang==14.0.6
librosa==0.9.1
llvmlite==0.38.0
Markdown==3.3.6
MarkupSafe==2.1.1
matplotlib==3.6.2
matplotlib-inline==0.1.3
mistune==0.8.4
more-itertools==9.0.0
multidict==6.0.2
multiprocess==0.70.12.2
munch==2.5.0
music21==8.1.0
nbclassic==0.3.7
nbclient==0.6.0
nbconvert==6.5.0
nbformat==5.3.0
nest-asyncio==1.5.5
notebook==6.4.11
notebook-shim==0.1.0
numba==0.55.1
numpy==1.21.6
oauthlib==3.2.0
omegaconf==2.1.2
opt-einsum==3.3.0
packaging==21.3
pandas==1.4.2
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.1.0
pooch==1.6.0
progressbar==2.5
prometheus-client==0.14.1
prompt-toolkit==3.0.29
protobuf==3.19.6
psutil==5.9.0
ptyprocess==0.7.0
pure-eval==0.2.2
pyarrow==7.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.21
pyDeprecate==0.3.2
Pygments==2.11.2
pyparsing==3.0.8
pyrsistent==0.18.1
python-dateutil==2.8.2
pytorch-fast-transformers==0.4.0
pytorch-lightning==1.6.1
pytz==2022.1
PyYAML==6.0
pyzmq==22.3.0
regex==2022.3.15
requests==2.27.1
requests-oauthlib==1.3.1
resampy==0.2.2
responses==0.18.0
rich==12.2.0
rotary-embedding-torch==0.1.5
rsa==4.8
sacremoses==0.0.49
scikit-learn==1.0.2
scipy==1.8.0
Send2Trash==1.8.0
six==1.16.0
sniffio==1.2.0
SoundFile==0.10.3.post1
soupsieve==2.3.2.post1
stack-data==0.2.0
tensorboard==2.11.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorboardX==2.5.1
tensorflow==2.11.0
tensorflow-estimator==2.11.0
tensorflow-io-gcs-filesystem==0.28.0
termcolor==2.1.1
terminado==0.13.3
threadpoolctl==3.1.0
tinycss2==1.1.1
tokenizers==0.12.1
torch==1.11.0
torchaudio==0.11.0
torchmetrics==0.8.0
torchtext==0.12.0
torchvision==0.12.0
tornado==6.1
TorToiSe==2.4.0
tqdm==4.64.0
traitlets==5.1.1
transformers==4.18.0
typing_extensions==4.2.0
Unidecode==1.3.4
urllib3==1.26.9
wcwidth==0.2.5
webcolors==1.12
webencodings==0.5.1
websocket-client==1.3.2
Werkzeug==2.1.1
widgetsnbextension==3.6.0
wrapt==1.14.1
xxhash==3.0.0
yarl==1.7.2
```

I hope that helps! I suspect it is the Hydra version, there are some weird breaking changes in the newer versions of Hydra that aren't compatible with how I specified the configs here.

**Comment by dawson-miller** [comment]
Thank you for this list. I don't think it is the hydra version. Based on the requirements, I first had to install the referenced torch versions, but the installation still broke at Tortoise and then again at pytorch-fast-transformers, so I removed them from the list.

Could you try making a fresh environment (or preferably a docker container) and testing the training code? I see packages in the list you sent that aren't utilized in train.py such as numba. Hopefully the OS version doesn't play a role, but a docker container would control for that. For reference I am using Ubuntu 22.04 in WSL.

**Comment by RF5** [comment]
I see. Here is the steps I took on a fresh linux machine to run training:

- Install python 3.10.13 with miniconda
- Clone repos
- Run `pip install torch torchaudio einops opt_einsum scipy fastprogress omegaconf pandas`
- I had to further install `pip install tensorboard packaging`. I have updated these packages in the readme.
- Install sox (if not already installed). For ubuntu there is a simple apt package. Or using anaconda: `conda install -c conda-forge sox`. This is a dependency of torchaudio, and is needed to load audio. 
- `python setup.py install` to compile cauchy kernel extension from the state-spaces repo.
- Run the training line as in the readme: `python train.py train_csv=splits/sc09-train.csv valid_csv=splits/sc09-valid.csv checkpoint_path=runs/run1/ validation_interval=2500`

This seems to train fine on my local machine? I have attached my fresh conda env dump as well below:

[env.txt](https://github.com/RF5/simple-sashimi/files/13423354/env.txt)

Hope that helps!
